src/NuGet.Core/NuGet.Protocol/Resources/VulnerabilityInfoResourceV3.cs

```diff
@@ -46,14 +45,10 @@ public async Task<IReadOnlyList<V3VulnerabilityIndexEntry>> GetVulnerabilityFile
             try
             {
                 vulnFiles = await httpSourceResource.HttpSource.GetAsync(request,
-                    result =>
+                    async result =>
                     {
-                        using (var textReader = new StreamReader(result.Stream))
-                        using (var jsonReader = new JsonTextReader(textReader))
-                        {
-                            var parsed = JsonExtensions.JsonObjectSerializer.Deserialize<IReadOnlyList<V3VulnerabilityIndexEntry>>(jsonReader);
-                            return Task.FromResult(parsed);
-                        }
+                        var parsed = await JsonSerializer.DeserializeAsync<IReadOnlyList<V3VulnerabilityIndexEntry>>(result.Stream, JsonExtensions.JsonSerializerOptions);
```

Comment:
What's our policy on using `return await`?  Or is this just for ease of debugging?  Just curious, as my preference is `return await`.

Reply:
Personally, I hate `return await`, because I want to optimize code for debuggability. There's no runtime performance penalty.

But I don't remember anything in our repo coding guidelines about it one way or another.